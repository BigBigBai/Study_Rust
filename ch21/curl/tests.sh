cargo clean
cargo build
(cargo run -- "https://www.eecg.toronto.edu/~bli/ece1724/assignments/files/lab3.html"
cargo run -- "www.eecg.toronto.edu"
cargo run -- "data://www.eecg.toronto.edu"
cargo run -- "http//www.eecg.toronto.edu"
cargo run -- "https://[...1]"
cargo run -- "https://255.255.255.256"
cargo run -- "http://127.0.0.1:65536"
cargo run -- "https://example.rs"
cargo run -- "https://www.eecg.toronto.edu/~bli/ece1724/assignments/files/lab4.html"
cargo run -- "https://jsonplaceholder.typicode.com/posts" -d "userId=1&title=Hello World" -X POST
cargo run -- --json '{"title": "World", "userId": 5}' "https://dummyjson.com/posts/add"
cargo run -- --json '{"title": "World"; "userId": 5}' "https://dummyjson.com/posts/add"
)2>&1 | grep -v -E 'Compiling|Running|Finished' > tests.log
diff tests.log tests.expected