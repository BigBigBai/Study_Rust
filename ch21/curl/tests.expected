Requesting URL: https://www.eecg.toronto.edu/~bli/ece1724/assignments/files/lab3.html
Method: GET
Response body:
<html>
<body>
<h1>
Hello, World!
</h1>
</body>
</html>
Requesting URL: www.eecg.toronto.edu
Method: GET
Error: The URL does not have a valid base protocol.
Requesting URL: data://www.eecg.toronto.edu
Method: GET
Error: The URL does not have a valid base protocol.
Requesting URL: http//www.eecg.toronto.edu
Method: GET
Error: The URL does not have a valid base protocol.
Requesting URL: https://[...1]
Method: GET
Error: The URL contains an invalid IPv6 address.
Requesting URL: https://255.255.255.256
Method: GET
Error: The URL contains an invalid IPv4 address.
Requesting URL: http://127.0.0.1:65536
Method: GET
Error: The URL contains an invalid port number.
Requesting URL: https://example.rs
Method: GET
Error: Unable to connect to the server. Perhaps the network is offline or the server hostname cannot be resolved.
Requesting URL: https://www.eecg.toronto.edu/~bli/ece1724/assignments/files/lab4.html
Method: GET
Error: Request failed with status code: 404.
Requesting URL: https://jsonplaceholder.typicode.com/posts
Method: POST
Data: userId=1&title=Hello World
Response body (JSON with sorted keys):
{
  "id": 101,
  "title": "Hello World",
  "userId": "1"
}
Requesting URL: https://dummyjson.com/posts/add
Method: POST
JSON: {"title": "World", "userId": 5}
Response body (JSON with sorted keys):
{
  "id": 252,
  "title": "World",
  "userId": 5
}
Requesting URL: https://dummyjson.com/posts/add
Method: POST
JSON: {"title": "World"; "userId": 5}
thread 'main' panicked at src/main.rs:84:58:
Invalid JSON: Error("expected `,` or `}`", line: 1, column: 18)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace